code/modules/mob/living/simple_animal/bot/firebot.dm

```diff
@@ -0,0 +1,306 @@
+			speech_cooldown = world.time
+
+		if(emagged == 2) // When emagged, knock people down and wet the floors under em
+			var/mob/living/M = target_fire
+			M.Knockdown(3)
+			src.do_attack_animation(M)
+			visible_message("<span class='danger'>[src] rams into [M] knocking them over!</span>")
+
+			var/turf/T = get_turf(M.loc)
+
+			if(isopenturf(T))
+				var/turf/open/theturf = T
+				theturf.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+		else
+			flick("firebot1_use", src)
+			spray_water(target_fire, src)
+
+		soft_reset()
+
+	// Target ran away
+	else if(target_fire && path.len && (get_dist(target_fire,path[path.len]) > 2))
+		path = list()
+		mode = BOT_IDLE
+		last_found = world.time
+
+	else if(target_fire && stationary_mode)
+		soft_reset()
+		return
+
+	if(target_fire && (get_dist(src, target_fire) > 2))
+
+		path = get_path_to(src, get_turf(target_fire), /turf/proc/Distance_cardinal, 0, 30, 1, id=access_card)
+		mode = BOT_MOVING
+		if(!path.len)
+			soft_reset()
+
+	if(path.len > 0 && target_fire)
+		if(!bot_move(path[path.len]))
+			old_target_fire = target_fire
+			soft_reset()
+		return
+
+	// We got a target but it's too far away from us
+	if(path.len > 8 && target_fire)
+		frustration++
+
+	if(auto_patrol && !target_fire)
+		if(mode == BOT_IDLE || mode == BOT_START_PATROL)
+			start_patrol()
+
+		if(mode == BOT_PATROL)
+			bot_patrol()
+
+
+//Look for burning people or turfs around the bot
+/mob/living/simple_animal/bot/firebot/process_scan(atom/scan_target)
+	var/result
+
+	if(scan_target == src)
+		return result
+
+	if(is_burning(scan_target))
+		if((detected_cooldown + DETECTED_VOICE_INTERVAL) < world.time)
+			speak("Fire detected!")
+			playsound(src.loc, "sound/voice/firebot/detected.ogg", 50, 0)
+			detected_cooldown = world.time
+		result = scan_target
+
+	return result
+
+/mob/living/simple_animal/bot/firebot/temperature_expose(datum/gas_mixture/air, temperature, volume)
+	if((temperature > T0C + 200 || temperature < BODYTEMP_COLD_DAMAGE_LIMIT) && foam_cooldown + FOAM_INTERVAL < world.time)
+		new /obj/effect/particle_effect/foam/firefighting(loc)
+		foam_cooldown = world.time
+	..()
+
+/mob/living/simple_animal/bot/firebot/proc/spray_water(atom/target, mob/user)
+	if(stationary_mode)
+		flick("firebots_use", user)
+	else
+		flick("firebot1_use", user)
+	internal_ext.afterattack(target, user, null)
+
+/mob/living/simple_animal/bot/firebot/update_icon()
+	if(!on)
+		icon_state = "firebot0"
+		return
+	if(IsStun() || IsParalyzed())
+		icon_state = "firebots1"
+		return
+	else if(stationary_mode) //Bot has yellow light to indicate stationary mode.
+		icon_state = "firebots1"
+	else
+		icon_state = "firebot1"
+
+
+/mob/living/simple_animal/bot/firebot/explode()
+	on = FALSE
+	visible_message("<span class='boldannounce'>[src] blows apart!</span>")
+
+	new /obj/item/assembly/prox_sensor(src.loc)
+	new /obj/item/clothing/head/hardhat/red(src.loc)
```

Comment:
use drop_location(). You had this all right the first time with `Tsec`, not sure why you keep changing it. Refer to the below code.

https://github.com/tgstation/tgstation/blob/a9ca93b60dbf7b9f0fc5cbe94c406a2126312fee/code/modules/mob/living/simple_animal/bot/floorbot.dm#L361-L379